connectors/agent/logger.py

```diff
@@ -0,0 +1,29 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License 2.0;
+# you may not use this file except in compliance with the Elastic License 2.0.
+#
+import logging
+
+import ecs_logging
+
+root_logger = logging.getLogger("agent_component")
+handler = logging.StreamHandler()
+handler.setFormatter(ecs_logging.StdlibFormatter())
+root_logger.addHandler(handler)
+root_logger.setLevel(logging.INFO)
+
+
+def get_logger(module):
+    logger = root_logger.getChild(module)
+
+    if logger.hasHandlers():
+        return logger
```

Comment:
It's actually a bit awkward.

This function operates on the assumption that only it can create logger and add handlers.

There's no way to get child logger and learn if it was just created, or it's already been created and we get the cached one back. 

One way people suggest is to check for some factor that can distinguish whether it was just created or not, and having handlers in our case is such a factor - if it was just created, there are no handlers and we should add them. If it has handlers, this file has created it and correct handlers should already be assigned.

I'm not super happy with it, I'm up to follow up with #python channel about it.

Reply:
https://stackoverflow.com/questions/53129716/how-to-check-if-a-logger-exists